clinics/merge/plan.go

```diff
@@ -0,0 +1,34 @@
+package merge
+
+import (
+	"context"
+)
+
+type Plan interface {
+	PreventsMerge() bool
+}
+
+type Planner[T Plan] interface {
```

Comment:
The only reason is to be able to run multiple planners and return a typed list of plans. This code would have been duplicated if the planner wasn't generic:
```
func RunPlanners[T Plan](ctx context.Context, planners []Planner[T]) ([]T, error) {
	result := make([]T, 0, len(planners))
	for _, planner := range planners {
		res, err := planner.Plan(ctx)
		if err != nil {
			return nil, err
		}
		result = append(result, res)
	}
	return result, nil
}
```

Reply:
Since the actual types do different things I would prefer a function over an interface and explicitly stating the return type for ease of reading. Something along the lines of:

```go
func runPlanners[T any](planners ...func(ctx context.Context) (T, error)) ([]T, error) {
	var res []T
	for _, runner := range planners {
		r, err := runner(context.Background())
		if err != nil {
			return nil, err
		}
		res = append(res, r)
	}
	return res, nil
}


type intermediatePlanner struct {
	SettingsPlanners []func(context.Context) (SettingsPlan, error)
	TagPlanners      []func(context.Context) (TagPlan, error)
	...
}

func (m *ClinicMergePlanner) Plan(ctx context.Context) (plan ClinicMergePlan, err error) {
	intermediate.SettingsPlanners = []func(context.Context) (SettingsPlan, error) {
		NewSettingsReporterPlanner(source, target, GetMRNRequiredSettings, TaskTypeClinicSettingsMRNRequired).Plan,
		NewSettingsReporterPlanner(source, target, GetMRNUniqueSettings, TaskTypeClinicSettingsMRNUnique).Plan,
		...
	}
}
```

but that might not be worth the effort at this point.